### PR TITLE
Build 8.1 beta

### PR DESCRIPTION
**Pull request by peastman**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

Most of this PR is just the latest changes from the main branch.  Aside from that, I just updated the version, build number, revision, and channel target.  I also added scipy as a runtime dependency since it's used by the benchmarking script (see https://github.com/openmm/openmm/issues/3938).

### CODE REVIEW

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

**Comment by peastman** [comment]
I'm not sure what's causing the merge conflicts.  It reports conflicts with changes that are older than 8.0.  Has the repo history somehow gotten messed up?

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

**Comment by peastman** [comment]
I hopefully resolved the merge conflicts, but I'm not at all sure I did everything correctly. @mikemhenry can you take a look?

**Comment by RaulPPelaez** [comment]
Seems like Python compilation fails:
```shell
[ 43%] Built target OpenMM
[ 43%] Building OpenMM Python binary module...
OPENMM_LIB_PATH = /home/conda/feedstock_root/build_artifacts/openmm_1696463697351/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib
OPENMM_INCLUDE_PATH = /home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/./include/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/./include/openmm/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/./include/openmm/internal/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/openmmapi/include/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/openmmapi/include/openmm/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/openmmapi/include/openmm/internal/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/olla/include/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/olla/include/openmm/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/olla/include/openmm/internal/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/serialization/include/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/serialization/include/openmm/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/serialization/include/openmm/internal/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/plugins/amoeba/openmmapi/include/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/plugins/amoeba/openmmapi/include/openmm/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/plugins/amoeba/openmmapi/include/openmm/internal/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/plugins/rpmd/openmmapi/include/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/plugins/rpmd/openmmapi/include/openmm/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/plugins/rpmd/openmmapi/include/openmm/internal/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/plugins/drude/openmmapi/include/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/plugins/drude/openmmapi/include/openmm/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/plugins/drude/openmmapi/include/openmm/internal
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/openmm_1696463697351/work/build/python/setup.py", line 10, in <module>
    from setuptools import setup
ModuleNotFoundError: No module named 'setuptools'
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -62,10 +62,15 @@ requirements:
     # OpenCL ICD
     - ocl-icd  # [linux]
     - khronos-opencl-icd-loader  # [win or (osx and opencl_impl == 'khronos')]
+    - libcufft-dev  # [(linux64 or ppc64le or win) and (cuda_compiler_version or "").startswith("12")]
+    - cuda-profiler-api  # [(linux64 or ppc64le or win) and (cuda_compiler_version or "").startswith("12")]
+    - cuda-nvrtc-dev  # [(linux64 or ppc64le or win) and (cuda_compiler_version or "").startswith("12")]
+    - cuda-driver-dev  # [(linux64 or ppc64le or win) and (cuda_compiler_version or "").startswith("12")]
```
Comment (RaulPPelaez) [inline recipe/meta.yaml:68]: Maybe pip/setuptools must be included as a host dependency?

**Comment by peastman** [comment]
I'll try that.  I don't understand, because it lists setuptools as being installed:

```
2023-10-04T23:54:55.0732315Z setuptools                68.2.2             pyhd8ed1ab_0    conda-forge
```

And the Mac builds pass, but the Linux and Windows ones don't.

**Comment by peastman** [comment]
All builds now pass.

**Comment by mikemhenry** [comment]
Awesome! Once this gets merged in and the builds post, I can spin up some CI jobs in the downstream ecosystem to test things.

**Comment by peastman** [comment]
Does it look ok to merge?

**Comment by mikemhenry** [comment]
I will give CI a few more minutes but I might have to kick it with an emtpy commit

**Comment by mikemhenry** [comment]
Just gave it a kick and looks like it is building now

**Comment by peastman** [comment]
It looks like there are no builds for Python 3.12.

**Comment by jakirkham** [comment]
The [Python 3.12 migrator isn't in the `rc` branch]( https://github.com/conda-forge/openmm-feedstock/tree/rc/.ci_support/migrations ). The migrator would needed to be added manually to enable those builds ( for example: https://github.com/conda-forge/openmm-feedstock/pull/109 ). Alternatively could enable bot migrations on the `rc` branch by adding it to [`bot/abi_migration_branches` in `conda-forge.yml`]( https://conda-forge.org/docs/maintainer/conda_forge_yml.html#bot )

**Comment by peastman** [comment]
If I manually merge the changes from #109 into the rc branch, will that do it correctly?

**Comment by jakirkham** [comment]
Would cherry-pick this commit ( https://github.com/conda-forge/openmm-feedstock/pull/109/commits/ce98e93b7765ee5379e9663e00ed3ff45586a8d9 ) and re-render. The `build/number` may also need to be updated

**Comment by peastman** [comment]
Thanks!  I just created a PR with that change.

**Comment by jakirkham** [comment]
xref: https://github.com/conda-forge/openmm-feedstock/pull/113

**Pull request merged**